File: GoNorthCS/Item.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GoNorth
{
    //------------------------------------------------------------------------------------------------
    public class Item
    {
        int _wordId;
        string _inventoryDescription;
        string _locationDescription;
        public int WordId { get => _wordId; }
        public string InventoryDescription { get => _inventoryDescription; }
        public string LocationDescription { get => _locationDescription; }

        //------------------------------------------------------------------------------------------------
        public Item(int wordId, string inventoryDescription, string locationDescription)
        {
            _wordId = wordId;
            _inventoryDescription = inventoryDescription;
            _locationDescription = locationDescription;
        }
    }

    //------------------------------------------------------------------------------------------------
    public class Inventory 
    {
        SortedSet<int> _intentoryItems = new SortedSet<int>();

        //------------------------------------------------------------------------------------------------
        public void AddItem(int WordId)
        {
            _intentoryItems.Add(WordId);
        }

        //------------------------------------------------------------------------------------------------
        public void RemoveItem(int WordId)
        {
            _intentoryItems.Remove(WordId);
        }

        //------------------------------------------------------------------------------------------------
        public bool HasItem(int WordId)
        {
            return _intentoryItems.Contains(WordId);
        }

        //------------------------------------------------------------------------------------------------
        public int ItemCount
        {
            get { return _intentoryItems.Count; }
        }

        //------------------------------------------------------------------------------------------------
        public void PrintItemsInLocation(Game game)
        {
            foreach( int WordId in _intentoryItems)
            {
                game.WriteOutput(game.GetItem(WordId).LocationDescription + "\n");
            }
        }

        //------------------------------------------------------------------------------------------------
        public void PrintItemsInInventory(Game game)
        {
            foreach (int WordId in _intentoryItems)
            {
                game.WriteOutput(game.GetItem(WordId).InventoryDescription + "\n");
            }
        }
    }
}

File: GoNorth/Item.cpp
//------------------------------------------------------------------------------------------------
// Item.cpp
//
// Implements CItem and CInventory classes
//------------------------------------------------------------------------------------------------

#include "stdafx.h"

namespace GoNorth
{
    //------------------------------------------------------------------------------------------------
    void CInventory::AddItem(WordId itemWordId)
    {
        // Insert item pointer into map
        m_inventoryItems.insert(itemWordId);
    }

    //------------------------------------------------------------------------------------------------
    void CInventory::RemoveItem(WordId itemWordId)
    {
        // Remove the item from the map
        m_inventoryItems.erase(itemWordId);
    }

    //------------------------------------------------------------------------------------------------
    bool CInventory::HasItem(WordId itemWordId) const
    {
        return m_inventoryItems.end() != m_inventoryItems.find(itemWordId);
    }

    //------------------------------------------------------------------------------------------------
    void CInventory::PrintItemsInInventory() const
    {
        for (auto itemWordId : m_inventoryItems)
        {
            CGame::Game().OutputStream() << CGame::Game().GetItem(itemWordId)->GetInventoryDescription() << "\n";
        }
    }

    //------------------------------------------------------------------------------------------------
    void CInventory::PrintItemsInLocation() const
    {
        for (auto itemWordId : m_inventoryItems)
        {
            CGame::Game().OutputStream() << CGame::Game().GetItem(itemWordId)->GetLocationDescription() << "\n";
        }
    }
}